index/generator/schema/schema.go

```diff
@@ -264,4 +266,16 @@ type Version struct {
 	DeploymentScopes map[DeploymentScopeKind]bool `yaml:"deploymentScopes,omitempty" json:"deploymentScopes,omitempty"`
 	Resources        []string                     `yaml:"resources,omitempty" json:"resources,omitempty"`
 	StarterProjects  []string                     `yaml:"starterProjects,omitempty" json:"starterProjects,omitempty"`
+	LastModified     string                       `yaml:"lastModified,omitempty" json:"lastModified,omitempty"`
```

Comment:
I think we might need to update the `schema.go` [here](https://github.com/devfile/registry-support/blob/ce8b9b9bed12d4dbafa1726e6ea490f2e3cf1ce7/index/generator/schema/schema.go#L25-L117)

Also after building and testing the new `devfile-index-schema` I've noticed that the `lastModified` is only added to versions. I think it should also be included in the [general schema](https://github.com/devfile/registry-support/blob/ce8b9b9bed12d4dbafa1726e6ea490f2e3cf1ce7/index/generator/schema/schema.go#L144-L166). For versioned stacks it should take the value of the `default` stack version. WDYT?

Reply:
I think that's a good point here. The "lastModified" for the stack should be the actual "lastModified" so after a second thought the stack `lastModified` should have the value of the latest modified version. WDYT?

Reply:
If we attach `lastModified` to the entire stack, what are we looking for in terms of the logic? If the stack shares a `lastModified` date with the latest modified version and that date is sent as a parameter it will send the same data back as it is configured now. Is the goal of attaching the `lastModified` to the entire stack as well just to include that information for the viewer since the response in terms of stack versions is going to be the same

Reply:
I implemented the change we talked about above, just trying to work through why the test for checking the generated index is failing. Jsondiff is saying they are equal but still failing it

Reply:
> If we attach `lastModified` to the entire stack, what are we looking for in terms of the logic? If the stack shares a `lastModified` date with the latest modified version and that date is sent as a parameter it will send the same data back as it is configured now. Is the goal of attaching the `lastModified` to the entire stack as well just to include that information for the viewer since the response in terms of stack versions is going to be the same

My main point for this is to try to keep consistency in the structure of the `v2index/all` call. That said when someone cass the endpoint they get:

```
stack:
{
    "name": "code-with-quarkus",
    "displayName": "Basic Quarkus",
    "description": "A simple Hello World Java application using Quarkus",
    "type": "sample",
    "tags": [
      "Java",
      "Quarkus"
    ],
    "icon": "https://design.jboss.org/quarkus/logo/final/SVG/quarkus_icon_rgb_default.svg",
    "projectType": "quarkus",
    "language": "java",
    "versions": [
      {
        "version": "1.1.0",
        "schemaVersion": "2.2.0",
        "default": true,
        "git": {
          "remotes": {
            "origin": "https://github.com/devfile-samples/devfile-sample-code-with-quarkus.git"
          }
        },
        "description": "java quarkus with devfile v2.2.0",
        "lastModified": "2024-04-29"
      },
      {
        "version": "1.0.0",
        "schemaVersion": "2.0.0",
        "git": {
          "remotes": {
            "origin": "https://github.com/elsony/devfile-sample-code-with-quarkus.git"
          }
        },
        "description": "java quarkus with devfile v2.0.0",
        "lastModified": "2024-02-15"
      }
    ]
}

sample:
{
  "name": "java-springboot-basic",
  "displayName": "Basic Spring Boot",
  "description": "A simple Hello World Java Spring Boot application using Maven",
  "type": "sample",
  "tags": [
    "Java",
    "Spring"
  ],
  "icon": "https://raw.githubusercontent.com/devfile-samples/devfile-stack-icons/main/spring.svg",
  "projectType": "springboot",
  "language": "java",
  "git": {
    "remotes": {
      "origin": "https://github.com/devfile-samples/devfile-sample-java-springboot-basic.git"
    }
  }
}
```
I'm thinking from a client perspective (having as an example the `registry-viewer`) it might be useful to include this information to all the items that someone is fetching through the call.

For the community registry this might not be a problem as [all of our samples have versions](https://github.com/devfile/registry/blob/main/extraDevfileEntries.yaml) but in case a registry has samples without versions the `lastModified` info won't be available to its viewer.

Reply:
A change may need to be done [here](https://github.com/devfile/registry/blob/main/.ci/generate_last_mod_file.sh) to get the `lastModified` of a sample that has no versions set unless we want to have a default value (if no versions).

Reply:
@thepetk with the most recent changes you might see that if a sample has no versions then the `lastModified` date attached to it is the default `time.Time` value. If we want to set that to be the last time the sample was modified we will need to edit the script [here ](https://github.com/devfile/registry/blob/main/.ci/generate_last_mod_file.sh) I believe. This PR is quite large I'm not sure if we want to incorporate that here or open a follow up issue?

Reply:
Yeah +1 and let's take it on a discussion, because I feel there are multiple approaches here

Reply:
After checking the latest, the non-versioned field works :) One thing we might also need to think about is if we want to serve a specific timezone & format, as we currently serve multiple formats.

For example I see:
* `2023-04-08T11:51:08Z`
* `2024-05-13T12:32:02+02:00`